nim_status/lib/settings.nim

```diff
@@ -0,0 +1,245 @@
+    db.exec("UPDATE settings SET node_config = ? WHERE synthetic_id = 'id'", ($value))
+  of SettingsType.PinnedMailservers:
+    db.exec("UPDATE settings SET pinned_mailservers = ? WHERE synthetic_id = 'id'", ($value))
+  of SettingsType.StickersPacksInstalled:
+    db.exec("UPDATE settings SET stickers_packs_installed = ? WHERE synthetic_id = 'id'", ($value))
+  of SettingsType.StickersPacksPending:
+    db.exec("UPDATE settings SET stickers_packs_pending = ? WHERE synthetic_id = 'id'", ($value))
+  of SettingsType.StickersRecentStickers:
+    db.exec("UPDATE settings SET stickers_recent_stickers = ? WHERE synthetic_id = 'id'", ($value))
+  of SettingsType.Usernames:
+    db.exec("UPDATE settings SET usernames = ? WHERE synthetic_id = 'id'", ($value))
+  of SettingsType.WalletVisibleTokens:
+    db.exec("UPDATE settings SET wallet_visible_tokens = ? WHERE synthetic_id = 'id'", ($value))
+  else: 
+    raise (ref SettingsError)(msg: "Setting: '" & $setting & "' cannot be updated or invalid data type received")
+
+
+proc saveSetting*(db: DbConn, setting: SettingsType, value: Address) =
+  case setting
+  of SettingsType.DappsAddress:
+    db.exec("UPDATE settings SET dapps_address = ? WHERE synthetic_id = 'id'", $value)
+  of SettingsType.EIP1581Address:
+    db.exec("UPDATE settings SET eip1581_address = ? WHERE synthetic_id = 'id'", $value)
+  else: 
+    raise (ref SettingsError)(msg: "Setting: '" & $setting & "' cannot be updated or invalid data type received")
+
+
+proc saveSetting*(db: DbConn, setting: string, value: auto) =
+  var s: SettingsType
+  try: 
+    s = parseEnum[SettingsType](setting)
+  except:
+    raise (ref SettingsError)(msg: "Unknown setting: '" & $setting)
+
+  saveSetting(db, s, value)
+
+
+proc getNodeConfig*(db: DbConn): JsonNode =
+  let query = "SELECT node_config FROM settings WHERE synthetic_id = 'id'"
+  for r in rows(db, query):
+    return r[0].strVal.parseJson
+
+
+proc getSettings*(db: DbConn): Settings =
+  let query = """SELECT address, chaos_mode, currency, current_network,
+  custom_bootnodes, custom_bootnodes_enabled, dapps_address, eip1581_address,
+  fleet, hide_home_tooltip, installation_id, key_uid, keycard_instance_uid,
+  keycard_paired_on, keycard_pairing, last_updated, latest_derived_path, log_level,
+  mnemonic, name, networks, notifications_enabled, push_notifications_server_enabled,
+  push_notifications_from_contacts_only, remote_push_notifications_enabled,
+  send_push_notifications, push_notifications_block_mentions, photo_path,
+  pinned_mailservers, preferred_name, preview_privacy, public_key, remember_syncing_choice,
+  signing_phrase, stickers_packs_installed, stickers_packs_pending, stickers_recent_stickers,
+  syncing_on_mobile_network, use_mailservers, usernames, appearance, wallet_root_address,
+  wallet_set_up_passed, wallet_visible_tokens, waku_bloom_filter_mode, waku_enabled,
+  webview_allow_permission_requests FROM settings WHERE synthetic_id = 'id'"""
```

Comment:
Wouldn't a `SELECT *` be simpler here?

Reply:
Yes, but then I do not know the order of the columns in order to populate the object attributes. (I could not find in sqlite docs if it's possible to refer to column names in the results of the query)